numpy/core/src/multiarray/methods.c

```diff
@@ -2416,6 +2416,18 @@ array_complex(PyArrayObject *self, PyObject *NPY_UNUSED(args))
     return c;
 }
 
+static PyObject *
+array_keys(PyArrayObject *self, PyObject *NPY_UNUSED(args))
+{
+    PyArray_Descr *descr = PyArray_DESCR(self);
+    PyObject *fields = descr->fields;
+    if (fields == NULL || fields == Py_None) {
```

Comment:
Documentation seems to suggest this is null, but I was getting it as `Py_None` instead. Best to be safe here and cover both cases, right?